From e4cde335d1771863a60b6931e51357b8470e85c4 Mon Sep 17 00:00:00 2001
From: Daniel Golle <daniel@makrotopia.org>
Date: Sun, 10 Dec 2023 22:41:39 +0000
Subject: [PATCH] ASoC: mediatek: mt7986: silence error in case of
 -EPROBE_DEFER

If probe is defered no error should be printed. Mute it.

Signed-off-by: Daniel Golle <daniel@makrotopia.org>
---
 sound/soc/mediatek/mt7986/mt7986-wm8960.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

--- a/sound/soc/mediatek/mt7986/mt7986-wm8960.c
+++ b/sound/soc/mediatek/mt7986/mt7986-wm8960.c
@@ -144,7 +144,9 @@ static int mt7986_wm8960_machine_probe(s
 
 	ret = devm_snd_soc_register_card(&pdev->dev, card);
 	if (ret) {
-		dev_err(&pdev->dev, "%s snd_soc_register_card fail: %d\n", __func__, ret);
+		if (ret != -EPROBE_DEFER)
+			dev_err(&pdev->dev, "%s snd_soc_register_card fail: %d\n", __func__, ret);
+
 		goto err_of_node_put;
 	}
 
